src/App.js

```diff
@@ -33,8 +34,10 @@ const App = () => {
   const [userAction, setUserAction] = useState('');
   const [errorOccurred, setErrorOccurred] = useState(false);
   const [lastDeleted, setLastDeleted] = useState({});
+  const [expenseToEdit, setExpenseToEdit] = useState({});
 
   // states to toggle the modals
+  const [editExpenseModalIsOpen, setEditExpenseModalIsOpen] = useState(false);
   const [confirmMonthModalIsOpen, setConfirmMonthModalIsOpen] = useState(false);
   const [confirmDeleteModalIsOpen, setConfirmDeleteModalIsOpen] =
     useState(false);
```

Comment:
remind me: what was our decision on these individual `*IsOpen` variables?

Reply:
Now that we've refactored the alerts, the modals should be due for a similar redesign. Since we have the `userAction` state variable now, we should be able to open and close any modal with a single `isModalOpen` variable. Essentially we'd use `userAction` to tell the app which modal and which alert to fire when a specific action is performed by the user.

Does this sound like a reasonable approach?

Reply:
i would like to encourage you to play around with a few different options. this is an interesting problem and i am sure it will come up again in a similar shape later. i do not have a strong preference on which option you pick. let me know if i can assist.

Reply:
I'll open a separate issue for this.